Chp12Lsn7/UnitTestProject/UnitTest.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Chp12Lsn7;
+using UserCommunication;
+
+namespace UnitTestProject
+{
+    [TestClass]
+    public class UnitTest
+    {
+        [TestMethod]
+        [ExpectedException(typeof(System.ArgumentException), "Invalid Input! Cannot enter a number less than or equal to zero.")]
+        public void SqrtTest()
+        {
+            int[] invalidNumbers = { 0, -2 };
+            int numberOfColumns = invalidNumbers.GetLength(0);
+
+            for (int i = 0; i < numberOfColumns; i++)
+            {
+                int testNumbers = invalidNumbers[i];
+                Program.CalculateSquareRoot(testNumbers);
+            }
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(System.ArgumentException), "Invalid Input! Cannot enter a number less than or equal to zero.")]
+        public void UserInputTestArgument()
+        {
+            string[] invalidNumbers = { "0", "-2" };
```

Comment:
This test is broken because it will throw an exception as soon as it processes the "0" argument, which the test will confirm is the correct response, but then the test will halt; it will never try to execute the next argument in the array which is "-2".